Wikipedia/Code/WMFNearbySectionController.m

```diff
@@ -169,7 +169,15 @@ - (BOOL)hasResults {
 - (void)nearbyViewModel:(WMFNearbyViewModel*)viewModel didFailWithError:(NSError*)error {
     self.nearbyError = error;
     [self.delegate controller:self didSetItems:self.items];
-    [self.delegate controller:self didFailToUpdateWithError:error];
+    
+    //This means there were 0 results - not neccesarily a "real" error.
+    //Only inform the delegate if we get a real error.
+    if (!([error.domain isEqualToString:MTLJSONAdapterErrorDomain] && error.code == MTLJSONAdapterErrorInvalidJSONDictionary)){
+        [self.delegate controller:self didFailToUpdateWithError:error];
+    }
+
+    //Don't try to update after we get an error.
+    [self.viewModel stopUpdates];
```

Comment:
not sure stopping updates is a good idea because:

- we might get valid (or non-empty) results w/ a new location
- we might be able to determine the location if it's initially unknown (which is a specific error)
- if we have results already, we want to continue updating distance & heading